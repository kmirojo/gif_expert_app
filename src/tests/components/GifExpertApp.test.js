import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../../GifExpertApp";

describe('<GifExpertApp />', () => {
    const wrapper = shallow(
        <GifExpertApp />
    );

    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should display a category list', () => {
        const categories = ["Man of Steel", "Dark Knight"];
        const wrapper = shallow(
            <GifExpertApp defaultCategories={categories}/>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find("GifGrid") ).toHaveLength(2);

    })
    
})
