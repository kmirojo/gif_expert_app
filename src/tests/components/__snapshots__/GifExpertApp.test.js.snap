// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> should display a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Man of Steel"
      key="Man of Steel"
    />
    <GifGrid
      category="Dark Knight"
      key="Dark Knight"
    />
  </ol>
</Fragment>
`;

exports[`<GifExpertApp /> should match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
